spec/card_generator_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require './lib/card_generator'
+
+RSpec.describe CardGenerator do
+  # Uses before method to repeat code below for each it method
+  before(:each) do
+    @filename = 'cards.txt'
+    @generator = CardGenerator.new(@filename)
+    @cards = @generator.cards
+  end
+
+  it 'exists' do
+    expect(@generator).to be_instance_of(CardGenerator)
+  end
+
+  it 'file exists' do
+    expect(File.exist?(@filename)).to eq(true)
+  end
+
+  it 'file is a file' do
+    expect(File.file?(@filename)).to eq(true)
+  end
+
+  it 'file is txt file' do
+    expect(File.extname(@filename)).to eq('.txt')
+  end
+
+  it 'cards == cards' do
+    expect(@cards).to eq(@cards)
```

Comment:
A better test here might assert that `@cards` is an `Array` and that the array contains `Card` objects. Right now you are just testing that an object (`@cards`) is equal to itself (`@cards`) which will always be true regardless of the code in your `CardGenerator` class.

Reply:
I would also probably check the size of the array. If you know `cards.txt` should've created 6 cards, we'd want to verify that as well.